subprojects/resources-s3/src/main/java/org/gradle/internal/resource/transport/aws/s3/S3Client.java

```diff
@@ -132,6 +147,48 @@ public void put(InputStream inputStream, Long contentLength, URI destination) {
         }
     }
 
+    private void putMultiPartObject(InputStream inputStream, Long contentLength, URI destination) {
+        try {
+            S3RegionalResource s3RegionalResource = new S3RegionalResource(destination);
+            String bucketName = s3RegionalResource.getBucketName();
+            String s3BucketKey = s3RegionalResource.getKey();
+            configureClient(s3RegionalResource);
+            List<PartETag> partETags = new ArrayList<>();
+            InitiateMultipartUploadRequest initRequest = new InitiateMultipartUploadRequest(bucketName, s3BucketKey)
+                .withCannedACL(CannedAccessControlList.BucketOwnerFullControl);
+            InitiateMultipartUploadResult initResponse = amazonS3Client.initiateMultipartUpload(initRequest);
```

Comment:
Are any of these `Closable` and/or need to be closed?

Reply:
I haven't found anything in the documentation or examples that suggest one needs to close or dispose of anything other than via the abortMultipartUpload() method in the exception handler.